front/app/utils/participationMethodUtils.ts

```diff
@@ -74,3 +83,29 @@ export function getAllParticipationMethods(
     throw `Unknown process_type ${project.attributes.process_type}`;
   }
 }
+
+/** Given the project and its phases, returns whether the InputManager should be shown in the back office.
+ */
+export function showInputManager(
+  project: IProjectData,
+  phases?: Error | IPhaseData[] | null | undefined
+): boolean {
+  let showInputManager = false;
+  if (project.attributes.process_type === 'continuous') {
+    showInputManager = getMethodConfig(
+      project.attributes.participation_method
+    ).showInputManager;
+  } else if (project.attributes.process_type === 'timeline') {
+    if (!isNilOrError(phases)) {
+      phases.map((phase) => {
```

Comment:
It is not necessary to use `map` here. `forEach` would be good enough. But it is better to use `some`, like:

```
showInputManager = phases.some((phase) => getMethodConfig(phase.attributes.participation_method).showInputManager)
```
Another possible improvement is not to use the `showInputManager` variable, but to jump out of the function with an early return in each branch of the `if` statement. If you do that, there will be `if` statements below each other, without the need to use `else if`. It will improve readability, because there is less nesting of code.

Reply:
Agreed - using "map" didn't like jumping out of the function since not all paths returned a value when I used with an if statement. I'll use the "some" like you suggest instead. Thanks!!